### ISSUE TITLE
bdh Hangs intermittently

### ISSUE THREAD

**Issue by vishalg** [Rblp/Rblpapi#120]
I am trying to download some historical data using BDH. However, the calls to bdh hang (blocked waiting?) or fail with an error message "Failed to open //blp/refdata". When it hangs I eventually have to kill the R process. This has only started to happen recently and I can't pin point exactly what triggered it. A few more details

- Using a connection to localhost.
- There are multiple consecutive bdh invocations within the code. Each invocation downloads two years worth of daily historical data for c.25 data series. Code  usually does not hang on the first invocation rather one of the subsequent invocations tends to hang. I have also tried "Sleep"ing the R code for upto 30 seconds between bdh invocations which does not help.
- An excel session is open with blp and bdh calls, the spreadsheet is not on AutoCalc but is recalculated periodically at 5 minute intervals. I have not noticed any causality between the blocks and excel re-calculation, other than the fact that the hanging problem happens a lot less frequently when no excel sessions are open.
- Bloomberg terminal app is also running on the desktop where the code is being run.
- Similar behaviour with Rbbg package 
- Using the Bloomberg Java API directly does not seem to run into these issues.
- R version (3.1.0)
- Rblpapi version (0.3.2)
- OS (Windows 7)
- Killing the Bloomberg API processes helps, as does killing the excel spreadsheets.

**Comment by eddelbuettel** [comment]
Not reproducible.

**Comment by wmorgan85** [comment]
When this issue occurs next, can you try producing the same request using
the API Data Demo Tool (Start > Bloomberg > API Data demo). Check the log
results button and then send request. This will give a log that you can
send to the Bloomberg Helpdesk to troubleshoot further.

**Comment by vishalg** [comment]
Thanks. I'll try with the Bloomberg API Demo Tool. Also, will try to isolate the issue into something reproducible.

**Comment by eddelbuettel** [comment]
@vishalg  Appreciate that -- as this is indeed client/server with a 'nebulous' large infrastructure behind it this may be tricky.

But in fairness, our hands are tied unless we can reproduce something.

**Comment by vishalg** [comment]
@eddelbuettel @wmorgan85, I was able to track the issue a bit further using the Java API directly. For some reason, the Session is being closed and the call to nextEvent waits forever in the event loop. I got around that by invoking nextEvent with a TIMEOUT setting and then handling the TIMEOUT event. Which may be an idea to incorporate in Rblpapi.

However, the current issue has nothing to do with Rblpapi and therefore I'll mark it closed. Thanks for your help!
